themes/classic/templates/catalog/_partials/product-details.tpl

```diff
@@ -36,7 +36,7 @@
   {/block}
 
   {block name='product_availability_date'}
-    {if $product.availability_date}
+    {if $product.availability_date && ({$smarty.now|date_format:'%Y-%m-%d'} < $product.availability_date)}
```

Comment:
Okay, now this works.

However this should still be checked in a PHP class, as I said in the previous comment. If we don't do this, every theme maintainer will have to update their own.

Reply:
I saw that as a bug, not improvement or customization.

Reply:
it needs to be checked in PHP, you can't rely on date comparison from template, we just need.to set $product.availability_date to false if its needed